theme/icons.go

```diff
@@ -1,41 +1,56 @@
 package theme
 
-import "fyne.io/fyne"
+import (
+	"bytes"
+	"encoding/xml"
+	"log"
+
+	"fyne.io/fyne"
+)
 
 // ThemedResource is a resource wrapper that will return an appropriate resource
-// for the currently selected theme. In this implementation it chooses between a dark
-// and light alternative to match the current setting.
+// for the currently selected theme.
 type ThemedResource struct {
-	dark  fyne.Resource
-	light fyne.Resource
-}
-
-func isThemeLight() bool {
-	return fyne.CurrentApp().Settings().Theme().BackgroundColor() == lightBackground
+	source *fyne.StaticResource
 }
 
-// Name returns the underlrying resource name (used for caching)
+// Name returns the underlying resource name (used for caching)
 func (res *ThemedResource) Name() string {
-	if isThemeLight() {
-		return res.light.Name()
-	}
-
-	return res.dark.Name()
+	return res.source.Name()
 }
 
 // Content returns the underlying content of the correct resource for the current theme
 func (res *ThemedResource) Content() []byte {
-	if isThemeLight() {
-		return res.light.Content()
+	rdr := bytes.NewReader(res.source.Content())
+	clr := fyne.CurrentApp().Settings().Theme().IconColor()
+	s, err := svgFromXML(rdr)
+	if err != nil {
+		fyne.LogError("could not load SVG, falling back to static content:", err)
+		return res.source.Content()
 	}
-
-	return res.dark.Content()
+	if err := s.replaceFillColor(rdr, clr); err != nil {
+		fyne.LogError("could not replace fill color, falling back to static content:", err)
+		return res.source.Content()
+	}
+	b, err := xml.Marshal(s)
+	if err != nil {
+		fyne.LogError("could not marshal svg, falling back to static content:", err)
+		return res.source.Content()
+	}
+	return b
 }
 
 // NewThemedResource creates a resource that adapts to the current theme setting.
-// It is currently a simple pairing of a dark and light variant of the same resource.
-func NewThemedResource(dark, light fyne.Resource) *ThemedResource {
-	return &ThemedResource{dark, light}
+// TODO: In version 2.0 we need to change this signature to just accept a single StaticResource pointer
+func NewThemedResource(dark, light *fyne.StaticResource) *ThemedResource {
```

Comment:
I missed this sorry - we should still be using fyne.Resource no?

Reply:
yep, good catch, fixed